src/dev/jest/setup/mocks.js

```diff
@@ -35,6 +35,7 @@
  */
 
 jest.mock('ui/metadata');
+jest.mock('ui/new_platform');
```

Comment:
We intentionally didn't want to auto-mock the ui/new_platform in this file, and instead wanted tests to include `jest.mock('ui/new_platform')` it `ui/new_platform` should be auto-mocked, which we seem to also be doing. Why would it be better to put a generic `ui/new_platform` mock in place vs mocking specific services that depend on `ui/new_platform`?

Reply:
yeah I totally agree, I removed this line. This was temporary code that I accidentally committed while trying to debug why `jest.mock('ui/new_platform')` didn't work inside of tests (the problem ended up being related to `ui/new_platform` actually pointing to `ui/new_platform/index.ts`.)

Reply:
Not entirely sure what you mean with:
> Why would it be better to put a generic ui/new_platform mock in place vs mocking specific services that depend on ui/new_platform?

After the migration Plugins should preferably not pass Core deep into their component/object hierarchy and instead wrap Core in a service layer. This should prevent unit tests from having to mock all of core as it will be easy to identify and mock only the services being consumed by the unit under test. But while we're in this interim state with the migration under way, lots of plugins don't yet have this clean separation (and have angular code that makes this hard). This leads to tests breaking when all of core isn't mocked in a way that returns representative values (like Observables or Promises where these are expected).